pytest_mozwebqa/cloud/browserstack.py

```diff
@@ -0,0 +1,69 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from ConfigParser import ConfigParser
+import os
+
+import requests
+from selenium import webdriver
+
+
+class BrowserStack(object):
+
+    name = 'BrowserStack'
+
+    def __init__(self):
+        username = None
+        access_key = None
+
+        config = ConfigParser()
+        config.read('setup.cfg')
+
+        section = 'browserstack'
+        if config.has_section(section):
+            if config.has_option(section, 'username'):
+                username = config.get(section, 'username')
+            if config.has_option(section, 'access-key'):
+                access_key = config.get(section, 'access-key')
+
+        self.username = os.getenv('BROWSERSTACK_USERNAME', username)
+        self.access_key = os.getenv('BROWSERSTACK_ACCESS_KEY', access_key)
+
+        if self.username is None:
+            raise ValueError('BrowserStack username must be set!')
+        if self.access_key is None:
+            raise ValueError('BrowserStack access key must be set!')
+
+        self.username = username
```

Comment:
This doesn't seem correct. It looks like this will scrap the values found via `os.getenv` (which are stored in `self`) and just replace them with what was in the local variables before the environment check was even done.